### DISCUSSION TITLE
how is it possible to move components within a Canvas on button click without using drag drop ?

### DISCUSSION THREAD

**Discussion by NoumanAhmad-Braqsol** [GrapesJS/grapesjs#4019] [Q&A]
For example if there is a button icon in tool tab if i click on that current block move up and down
![sss](https://user-images.githubusercontent.com/72378934/146520088-924861f8-7f8d-4f4d-82a4-b23b659e4931.PNG)
I am using this function 

editor.Commands.add("move-up", {
        run: function (editor) {
          const selected = editor.getSelected().getEl();
          const selectedPrevious = selected.previousElementSibling;
         const parent = selected.parentNode;
          parent.insertBefore(selected, selectedPrevious); 
}})

It moves the componet in canvas .but when i do editor.getHtml();
It gives the original Html but in canvas now position is changed so editor.getHtml() should return the html after change

**Comment by artf** [comment]
You should update the [Component](https://grapesjs.com/docs/api/component.html) models, not the DOM.
